src/core/ext/filters/client_channel/lb_policy.h

```diff
@@ -92,14 +92,51 @@ class LoadBalancingPolicy : public InternallyRefCounted<LoadBalancingPolicy> {
     GRPC_ABSTRACT_BASE_CLASS
   };
 
+  /// Interface for access metadata.
+  class MetadataInterface {
```

Comment:
Sorry I have a rather stupid question: There would be significant overheads just because of this is an abstract class.  Looking at the usage, I cannot tell whether this interface would have other implementations.  Are you envisioning some other immediate implementation of the interface?

If not, would it make sense to eliminate the `Interface` and replace with `Metadata`?

Reply:
Thank you for the context and being patient with me offline! Makes sense now.